JS: implement SocketIO on top of the EventEmitter model (#2636)
Semmle/ql: change-notes/1.24/analysis-javascript.md

## Changes to libraries

* The predicates `RegExpTerm.getSuccessor` and `RegExpTerm.getPredecessor` have been changed to reflect textual, not operational, matching order. This only makes a difference in lookbehind assertions, which are operationally matched backwards. Previously, `getSuccessor` would mimick this, so in an assertion `(?<=ab)` the term `b` would be considered the predecessor, not the successor, of `a`. Textually, however, `a` is still matched before `b`, and this is the order we now follow.

> * A model has been added for the NodeJS `EventEmitter`. The `SocketIO` model has been refactored to use this model.

Reviewer: For the sake of consistency, this section should mention libraries.

How about:
```suggestion
* An extensible model of the `EventEmitter` pattern has been implemented. 
```
And then mention: Node.js, SocketIO, Electron above at "* Support for the following frameworks and libraries has been improved:"
